core-ui/src/components/Extensibility/helpers/index.js

```diff
@@ -89,6 +90,41 @@ export function createTemplate(api, namespace, scope) {
   return template;
 }
 
+export function getDefaultPreset(presets, emptyTemplate) {
+  if (!presets || !presets.length) return null;
+  const defaultPreset = presets.find(preset => preset.default === true);
+  return defaultPreset
+    ? merge({}, { value: emptyTemplate }, defaultPreset)
+    : null;
+}
+
+export function PreparePresets(resource, presets) {
+  const { t: tExt } = useGetTranslation();
+  const { t } = useTranslation();
+
+  if (!presets || !presets.length) return null;
+  let preparedPresets = [];
+  let hasDefaultPreset = false;
+
+  presets.forEach(preset => {
+    if (preset.default === true) hasDefaultPreset = true;
+
+    preparedPresets.push(
+      merge({}, { value: resource }, { ...preset, name: tExt(preset.name) }),
+    );
+  });
+
+  if (!hasDefaultPreset) {
+    preparedPresets.unshift({
+      name: t('common.create-form.default-preset'),
+      value: resource,
+    });
+  }
```

Comment:
I'm not sure if we should allow to specify a default preset. In my opinion, a user should always be able to come back to a clear template. For example, in Pizzas you've specified `Margharita` as a default one. If the user wants to come back to a clear template, he won't be able to do that, which can be very frustrating. In busola, we always have a `Default` preset where none field is prefilled.